source/common/grpc/google_grpc_creds_impl.cc

```diff
@@ -140,6 +140,11 @@ getGoogleGrpcChannelCredentials(const envoy::api::v2::core::GrpcService& grpc_se
   } else {
     credentials_factory = Registry::FactoryRegistry<GoogleGrpcCredentialsFactory>::getFactory(
         google_grpc_credentials_factory_name);
+    if (Registry::FactoryRegistry<GoogleGrpcCredentialsFactory>::isDeprecated(
```

Comment:
Wondering if this can be rolled into `getFactory`, I know it is non-trivial because `registry.h` has limited dependency and templated. Also I think you missed `Config::Utility::translateOpaqueConfig`.

Reply:
Is it true that we must call `registerFactory()` after created factory with `Config::Utility::translateOpaqueConfig()` ? @lizan